task_base.py

```diff
@@ -503,20 +504,22 @@ def wait(self):
 
     def update_ee_tasks(self):
         if self.ee_tasks:
-            # possible ee task states: READY, RUNNING, COMPLETED, FAILED, CANCELLED, UNKNOWN
-            statuses = ee.data.getTaskStatus(self.ee_tasks.keys())
-            print(statuses)
-            for s in statuses:
-                ee_task_state = s["state"]
-                ee_task_id = s["id"]
-
-                if ee_task_state in self.EEFINISHED:
-                    if ee_task_state == self.EEFAILED:
-                        self._failed_ee_tasks[ee_task_id] = s
-                    del self.ee_tasks[ee_task_id]
-                    # TODO: log and update self.status based on EECOMPLETED, EEFAILED, EECANCELLED, EEUNKNOWN
-                else:
-                    self.ee_tasks[s["id"]] = s
+            try:
+                # possible ee task states: READY, RUNNING, COMPLETED, FAILED, CANCELLED, UNKNOWN
+                statuses = ee.data.getTaskStatus(self.ee_tasks.keys())
+                print(statuses)
+                for s in statuses:
+                    ee_task_state = s["state"]
+                    ee_task_id = s["id"]
+
+                    if ee_task_state in self.EEFINISHED:
+                        if ee_task_state == self.EEFAILED:
+                            self._failed_ee_tasks[ee_task_id] = s
+                        del self.ee_tasks[ee_task_id]
+                    else:
+                        self.ee_tasks[s["id"]] = s
+            except ConnectionResetError:
```

Comment:
Does ConnectionResetError need to be imported

Reply:
Apparently it's a builtin, so I don't think so. I don't get an import error when running, either.

Reply:
So it is, haven't used that one